students/JerryH/Lesson03/mailroom.py

```diff
@@ -1,79 +1,93 @@
     return all_names
 
-def main( op = 0 ):
-    while op > 3 or op < 1:
-        op = prompt()
 
-        if op == THANK_YOU_OPT: #Append or list donors
-            #Add donor's name or list the exisiting donors
-            d_name = thank_you_prompt()
-            while d_name.lower() == "list": #provide a list of donors if "list" is entered
-                #Get each unique donor's name from the list
-                for each_name in sorted(get_all_donor_names(donors)):
-                    print("{}".format(each_name))
-                d_name = thank_you_prompt()
+def list_all_donor_names(donor_list):
+    for each_name in sorted(get_all_donor_names(donor_list)):
+        print("{}".format(each_name))
 
-            #print(d_name)
+
+def main():
+    op = 0
+    while op != 3:
+        op = prompt()
+
+        if op == THANK_YOU_OPT:  # Append or list donors
+            donor_name = None
+            while not donor_name:
+                donor_name = donor_name_prompt()
+                if donor_name.lower() == "list":
+                    donor_name = None
+                    list_all_donor_names(donors)
+            # print(donor_name)
             donation = float(input("Please enter the donation amount:\n"))
-            #donors.append([d_name, donation])
-            if d_name not in get_all_donor_names(donors):
-                #print(donors)
-                donors.append([d_name, [donation]])
+            # donors.append([donor_name, donation])
+            if donor_name not in get_all_donor_names(donors):
+                # print(donors)
+                donors.append([donor_name, [donation]])
             else:
-                #print(get_all_donor_names(donors).index(d_name))
-                donors[get_all_donor_names(donors).index(d_name)][1].append(donation)
+                # print(get_all_donor_names(donors).index(donor_name))
+                donors[get_all_donor_names(donors).index(donor_name)][1].append(donation)
             print("Thank You Email:  Thansk for the donation!\n\n")
-            main()
-        elif op == REPORT_OPT: #Create report
+            #main()
+        elif op == REPORT_OPT:  # Create report
             print("\nDonor Name           |  Total Given | Num Gifts | Average Gift")
             print("---------------------------------------------------------------\n")
-            report = [] #initialize report
+            report = []  # initialize report
 
-            #for each_name in get_all_donor_names(donors):
+            # for each_name in get_all_donor_names(donors):
             for each_donor in donors:
-                total_donation = 0 #initialize total donation from the same donor
-                #Get total donation from each donor
+                total_donation = 0  # initialize total donation from the same donor
+                # Get total donation from each donor
                 for i in range(len(each_donor[1])):
                     total_donation += each_donor[1][i]
```

Comment:
so here you can do:
`total_donation = sum(each_donor[1])`

https://docs.python.org/2/library/functions.html#sum